native/common/jp_classhints.cpp

```diff
@@ -642,6 +635,15 @@ class JPConversionObject : public JPConversion
 		JP_TRACE("assignable", assignable, oc->getCanonicalName(), cls->getCanonicalName());
 		match.type = (assignable ? JPMatch::_derived : JPMatch::_none);
 
+		// User has request a Java class to class conversion.  We must pass through check it.
+		if (!assignable)
```

Comment:
would it be hard to create a Python regression test for this one?

Reply:
The code that depends on this change does not appear in this branch so it appears untested.   I can try to replicate a similar situation artificially.  Though we don't currently have a way to delete a conversion once added, so I will need to find something that wasn't harmful.